launch/test/launch/actions/test_push_and_pop_environment.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the PopEnvironment and PushEnvironment action classes."""
+
+from launch import LaunchContext
+from launch.actions import PopEnvironment
+from launch.actions import PushEnvironment
+
+
+def test_push_and_pop_environment_constructors():
+    """Test the constructors for PopEnvironment and PushEnvironment classes."""
+    PopEnvironment()
+    PushEnvironment()
+
+
+def test_push_and_pop_environment_execute():
+    """Test the execute() of the PopEnvironment and PushEnvironment classes."""
+    context = LaunchContext()
+
+    # does not change empty state
+    context.environment.clear()
```

Comment:
This modifies `os.environ`, which can negatively effect other tests in the same pytest process relying on it.
This bug was discovered and fixed in https://github.com/ros2/launch/pull/652